src/Persistence/Array_.php

```diff
@@ -56,7 +56,10 @@ public function add($m, $defaults = [])
 
         if ($f = $m->hasField($m->id_field)) {
             if (!$f->type) {
-                $f->type = 'integer';
+                //$f->type = 'integer';
+                // have to re-add field to use proper Field class
+                $m->removeField($m->id_field);
+                $m->addField($m->id_field, ['type'=>'integer']);
```

Comment:
probably could be decided earlier, before field is added.

Reply:
Not really because ID field is added in Model automatically even before Model is added into Persistence.

Reply:
ok, lets have id field seed then, which is by default id_field_seed = ['Integer'] but could also be something else.

Reply:
Added id field seed property in Model class and removed $type property dependency from Array Persistence class.